modules/apps/object/object-api/src/main/java/com/liferay/object/system/model/listener/SystemObjectDefinitionMetadataModelListener.java

```diff
@@ -89,6 +92,22 @@ public void onAfterUpdate(T originalBaseModel, T baseModel)
 			(T)baseModel.clone());
 	}
 
+	@Override
+	public void onBeforeCreate(T model) throws ModelListenerException {
+		try {
+			ObjectDefinition objectDefinition =
+				_objectDefinitionLocalService.fetchObjectDefinitionByClassName(
+					_getCompanyId(model), _modelClass.getName());
+
```

Comment:
Check if the objectDefinition is null is onBeforeCreate and onBeforeUpdate methods